src/java.desktop/macosx/classes/com/apple/laf/AquaCaret.java

```diff
@@ -189,20 +189,25 @@ protected synchronized void damage(final Rectangle r) {
         // intersection of the caret rectangle and the component less the border, if any.
         final Rectangle caretRect = new Rectangle(x, y, width, height);
         final Border border = getComponent().getBorder();
-        if (border != null) {
-            final Rectangle alloc = getComponent().getBounds();
-            alloc.x = alloc.y = 0;
+        final Rectangle alloc = getComponent().getBounds();
+        alloc.x = alloc.y = 0;
+        if (border != null && border.isBorderOpaque()) {
             final Insets borderInsets = border.getBorderInsets(getComponent());
             alloc.x += borderInsets.left;
             alloc.y += borderInsets.top;
             alloc.width -= borderInsets.left + borderInsets.right;
             alloc.height -= borderInsets.top + borderInsets.bottom;
             Rectangle2D.intersect(caretRect, alloc, caretRect);
+            x = caretRect.x;
+            y = caretRect.y;
+            width = Math.max(caretRect.width, 1);
+            height = Math.max(caretRect.height, 1);
+        } else {
+            x = alloc.x;
+            y = alloc.y;
+            width = alloc.width;
+            height = alloc.height;
```

Comment:
Currently the test case tests the else branch (no border case), since the if branch is also modified, would it be better to test both cases - with and without border?

Reply:
That is already being tested by existing testcases in CI and if I am not wrong by SwingSet2 too..